middleware.js

```diff
@@ -11,6 +12,38 @@ const SN_REFERRER = 'sn_referrer'
 const SN_REFERRER_NONCE = 'sn_referrer_nonce'
 // key for referred pages
 const SN_REFEREE_LANDING = 'sn_referee_landing'
+// main domain
+const SN_MAIN_DOMAIN = new URL(process.env.NEXT_PUBLIC_URL)
+// territory paths that needs to be rewritten to ~subname
+// const SN_TERRITORY_PATHS = ['~', 'recent', 'top', 'post', 'edit']
+
+async function customDomainMiddleware (request, referrerResp, domain, subName) {
+  // clone the url to build on top of it
+  const url = request.nextUrl.clone()
+  // we need pathname, searchParams and origin
+  const { pathname, searchParams, origin } = url
+  // set the subname in the request headers
+  const reqHeaders = new Headers(request.headers)
+  reqHeaders.set('x-stacker-news-subname', subName)
+
+  // TEST
+  console.log('[domains] custom domain', domain, 'with subname', subName) // TEST
+  console.log('[domains] main domain', SN_MAIN_DOMAIN) // TEST
+  console.log('[domains] pathname', pathname) // TEST
+  console.log('[domains] searchParams', searchParams) // TEST
+  console.log('[domains] origin', origin) // TEST
+
+  // TODO: 1. handle auth sync
+  // TODO: 2. redirect ~subname to /
+  // TODO: 3. rewrite / to ~subname
```

Comment:
Can we add TODO 3 to this PR?

Without it, it feels like the PR doesn't do much.